src/rom.c

```diff
@@ -56,49 +68,58 @@ static const char *rom_sizes[] = {
     [0x54] = "1.5MByte (96 banks)",
 };
 
-static uint8_t *rom_bytes;
+static uint8_t* rom_bytes = NULL;
+static size_t rom_size;
 
-void rom_setup(uint8_t *rom_data) {
-    char title[17];
+/******************************************************
+ *** LOCAL METHODS                                  ***
+ ******************************************************/
 
-    memcpy(title, &rom_data[0x134], 16);
-    title[16] = '\0';
-    
+static void rom_setup(void) {
+    char title[ROM_TITLE_LEN];
+    memcpy(title, &rom_bytes[ROM_TITLE_START_ADDR], ROM_TITLE_LEN - 1);
+    title[ROM_TITLE_LEN - 1] = '\0';
     LOG_INFO("Rom title: %s", title);
 
-    uint8_t type = rom_data[0x147];
+    uint8_t type = rom_bytes[CARTRIDGE_TYPE_ADDR];
     LOG_INFO("Cartridge type: %s (%02x)", cartridge_types[type], type);
 
-    uint8_t romsize_index = rom_data[0x148];
-    LOG_INFO("Cartridge rom size: %s (%02x)", rom_sizes[romsize_index], romsize_index);
-
-    rom_bytes = rom_data;
+    uint8_t rom_size_index = rom_bytes[CARTRIDGE_SIZE_ADDR];
+    LOG_INFO("Cartridge rom size: %s (%02x)", rom_sizes[rom_size_index], rom_size_index);
 }
 
-ErrorCode rom_load(const char *filename) {
-    int f;
-    struct stat st;
+/******************************************************
+ *** EXPOSED METHODS                                ***
+ ******************************************************/
 
-    uint8_t *bytes;
-    size_t rom_size;
+void rom_init(const char *file_name) {
+    int f = open(file_name, O_RDONLY);
+    if (f == -1)
+		LOG_EXIT("Opening file %s failed: %s", file_name, strerror(errno));
 
-    f = open(filename, O_RDONLY);
-    if(f == -1)
-        return ERR_FILE_IO;
+	struct stat st;
+    if (fstat(f, &st) == -1)
+		LOG_EXIT("Retrieving information about file %s failed: %s", file_name, strerror(errno));
 
-    if(fstat(f, &st) == -1)
-        return ERR_FILE_IO;
+	rom_size = (size_t) st.st_size;
+	rom_bytes = mmap(NULL, rom_size, PROT_READ, MAP_PRIVATE, f, 0);
+    if (rom_bytes == MAP_FAILED)
+		LOG_EXIT("mmap for file %s failed: %s", file_name, strerror(errno));
 
-    rom_size = (size_t) st.st_size;
-
-    bytes = mmap(NULL, rom_size, PROT_READ, MAP_PRIVATE, f, 0);
-    if(!bytes)
-        return ERR_FAILURE;
+    rom_setup();
+}
 
-    rom_setup(bytes);
-    return ERR_SUCCESS;
+void rom_destroy(void) {
+	if (rom_bytes != NULL && munmap(rom_bytes, rom_size) == -1) {
+		// We cannot use LOG_EXIT here since it calls rom_destroy
+		LOG_FATAL("munmap failed: %s", strerror(errno));
+		fflush(stdout);
+		fflush(stderr);
+		SDL_Quit();
```

Comment:
When this function gets called from `main`, `SDL_Quit()` was already executed.

Reply:
Indeed, I noticed that too. I didn't want to make any dependencies between teardowns. In this case it's okay, because you can call it multiple times: https://discourse.libsdl.org/t/importance-of-sdl-quit/18329

Reply:
But this should be resolved once we have a common teardown function.

Reply:
I see, but shouldn't it generally be avoided to call the same function twice, with the second call not having any effect?

Reply:
Sometimes it implemented like that exactly for that purpose 🤔 But I'm not entirely sure tbf.
I'd agree with you, we should figure something out that avoids this as best as possible. As mentioned in above threads by a common teardown.

Reply:
Resolved by use of `atexit`.